src/inc/til/small_vector.h

```diff
@@ -829,20 +822,36 @@ namespace til
 
             // 1. We've resized the vector to fit an additional `count` items.
             //    Initialize the `count` items at the end by moving existing items over.
-            std::uninitialized_move(end() - displacement, end(), _uninitialized_begin() + (_size + count - displacement));
+            //
+            // This line is noexcept:
+            // It can't throw because of the earlier static_assert(std::is_nothrow_move_assignable_v<T>).
+            std::uninitialized_move(end() - displacement, end(), _uninitialized_begin() + (new_size - displacement));
             _size = new_size;
 
             // 2. Now that all `new_size` items are initialized we can move as many
             //    items towards the end as we need to make space for `count` items.
+            //
+            // This line is noexcept:
+            // It can't throw because of the earlier static_assert(std::is_nothrow_move_assignable_v<T>).
             const auto displacement_beg = begin() + offset;
+            const auto displacement_end = displacement_beg + displacement;
             std::move_backward(displacement_beg, displacement_beg + (moveable - displacement), end() - displacement);
 
             // 3. Destroy the displaced existing items, so that we can use
             //    std::uninitialized_*/std::construct_* functions inside `func`.
             //    This isn't optimal, but better than nothing.
-            std::destroy(displacement_beg, displacement_beg + displacement);
+            std::destroy(displacement_beg, displacement_end);
 
-            func(displacement_beg);
+            try
+            {
+                func(displacement_beg);
+            }
+            catch (...)

```

Comment:
Do you need to propagate the noexcept keyword from the ctor of T to the lambda, so you can then propagate it here, so that the try/catch is erased if T is nothrow constructable?

Reply:
Effectively, this:

```c++
            return _generic_insert(pos, 1, [&](auto&& it) noexcept(noexcept(T(value))) {
                std::construct_at(&*it, value);
            });
```

with the innermost expression updated for the closest approximate operation?

Reply:
I don't think so... Dropping exception handlers seems like a pretty important optimization target for a compiler, so any compiler worth its value should do this. 🤔

Here's how I tested MSVC in particular... I put the class into godbolt and added the following:
```cpp
extern "C" void foobar();

struct NoexceptConstructible {
    NoexceptConstructible(int v) noexcept : value(v) {}
    int value;
};

struct ExceptConstructible {
    ExceptConstructible(int v) noexcept(false) : value(v) {}
    int value;
};

void test_int(til::small_vector<int, 16>& v) {
    v.insert(v.end(), 123);
}

void test_noexcept(til::small_vector<NoexceptConstructible, 16>& v) {
    v.insert(v.end(), 123);
}

void test_except(til::small_vector<ExceptConstructible, 16>& v) {
    v.insert(v.end(), 123);
}
```

Then I added a call to `foobar()` inside that exception handler.
None of the test functions caused a call to `foobar()` in the assembly.